powerline.theme

```diff
@@ -437,4 +567,243 @@ __powerline_vcs_section() {
+    else
+        __powerline_section "T" "${POWERLINE_DISPLAY_COLOR[@]}"
+    fi
+
+    __powerline_section "$_POWERLINE_USERNAME" "${POWERLINE_USER_COLOR[@]}"
+
+    __powerline_section "${_POWERLINE_HOST_ICON}${_POWERLINE_HOSTNAME}" "${POWERLINE_HOST_COLOR[@]}"
+
+    __powerline_section "$_lp_powerline_path" "${POWERLINE_PATH_COLOR[@]}"
+
+    if _lp_dirstack; then
+        __powerline_section "${LP_MARK_DIRSTACK}${lp_dirstack}" "${POWERLINE_DIRSTACK_COLOR[@]}"
+    fi
+
+    if _lp_shell_level; then
+        __powerline_section "${LP_MARK_SHLVL}${lp_shell_level}" "${POWERLINE_SHLVL_COLOR[@]}"
+    fi
+
+    local lp_chroot
+    if _lp_chroot; then
+        __powerline_section "${POWERLINE_CHROOT_MARKER}${lp_chroot}" "${POWERLINE_CHROOT_COLOR[@]}"
+    fi
+
+    __powerpuff_end_terminator
+
+    local left_sections="${powerline_sections}"
+    powerline_sections=""
+
+    local lp_time lp_analog_time
+    if _lp_time; then
+        __powerline_section "$lp_time" "${POWERLINE_TIME_COLOR[@]}"
+    elif _lp_analog_time; then
+        __powerline_section "$lp_analog_time" "${POWERLINE_TIME_COLOR[@]}"
+    fi
+
+    __powerline_battery_section
+
+    local lp_load_color
+    if _lp_load_color; then
+        __powerline_formatted_section "$lp_load_color" "${POWERLINE_LOAD_COLOR[@]}"
+    fi
+
+    local lp_ram_color
+    if _lp_ram_color; then
+        __powerline_formatted_section "$lp_ram_color" "${POWERLINE_RAM_COLOR[@]}"
+    fi
+
+    local lp_disk_color
+    if _lp_disk_color; then
+        __powerline_formatted_section "$lp_disk_color" "${POWERLINE_DISK_COLOR[@]}"
+    fi
+
+    local lp_temperature_color
+    if _lp_temperature_color; then
+        __powerline_formatted_section "$lp_temperature_color" "${POWERLINE_TEMPERATURE_COLOR[@]}"
+    fi
+
+    local lp_wifi_signal_strength_color
+    if _lp_wifi_signal_strength_color; then
+        __powerline_formatted_section "$lp_wifi_signal_strength_color" "${POWERLINE_WIFI_STRENGTH_COLOR[@]}"
+    fi
+
+    local lp_detached_sessions lp_running_jobs lp_stopped_jobs jobs_string=
+    if _lp_detached_sessions; then
+        jobs_string="${lp_detached_sessions}d"
+    fi
+    if _lp_jobcount; then
+        if (( lp_running_jobs > 0 )); then
+            [[ -n $jobs_string ]] && jobs_string+="${POWERLINE_SPACER}${POWERLINE_SOFT_DIVIDER}${POWERLINE_SPACER}"
+            jobs_string+="${lp_running_jobs}&"
+        fi
+        if (( lp_stopped_jobs > 0 )); then
+            [[ -n $jobs_string ]] && jobs_string+="${POWERLINE_SPACER}${POWERLINE_SOFT_DIVIDER}${POWERLINE_SPACER}"
+            jobs_string+="${lp_stopped_jobs}z"
+        fi
+    fi
+    __powerline_section "$jobs_string" "${POWERLINE_JOBS_COLOR[@]}"
+    __powerpuff_end_terminator
+
+    local right_sections="${powerline_sections}\r" # FIXME spacing problem
+
+    lp_terminal_format "${POWERLINE_LINE_COLOR[@]}"
+    _lp_fill "$left_sections" "$right_sections" "${lp_terminal_format}─"
```

Comment:
This forever loops. I figure I'll let you figure out why, since you wrote `_lp_fill()`. It does this with no config file. It also does this with the `theme-preview.sh` script, so that might be an easier way to test.

Here is the `set -x` output (starting at this line):
```bash
+ _lp_fill '\[\]╭─\[\] T \[\]\[\] user \[\]\[\] \[\]~ \[\]\[\]' '\[\]\[\]\r' '\[\]─'
+ local 'left=\[\]╭─\[\] T \[\]\[\] user \[\]\[\] \[\]~ \[\]\[\]' 'right=\[\]\[\]\r' 'fillchars=\[\]─' splitends=1
+ local ret
+ __lp_strip_escapes '\[\]╭─\[\] T \[\]\[\] user \[\]\[\] \[\]~ \[\]\[\]'
+ shopt -q extglob
+ local _lp_no_extglob=true
+ shopt -s extglob
+ ret=╭─ T  user  ~ 
+ [[ -n true ]]
+ shopt -u extglob
+ ret=╭─ T  user  ~ 
+ shopt -q promptvars
+ ret=╭─ T  user  ~ 
+ ret=╭─ T  user  ~ 
+ local left_as_text=╭─ T  user  ~ 
+ __lp_strip_escapes '\[\]\[\]\r'
+ shopt -q extglob
+ local _lp_no_extglob=true
+ shopt -s extglob
+ ret='\r'
+ [[ -n true ]]
+ shopt -u extglob
+ ret='\r'
+ shopt -q promptvars
+ ret='\r'
+ ret='\r'
+ local 'right_as_text=\r'
+ local gap_width=137
+ [[ 137 -lt 0 ]]
+ local filled=
+ __lp_strip_escapes '\[\]─'
+ shopt -q extglob
+ local _lp_no_extglob=true
+ shopt -s extglob
+ ret=─
+ [[ -n true ]]
+ shopt -u extglob
+ ret=─
+ shopt -q promptvars
+ ret=─
+ ret=─
+ local fillchars_as_text=─
+ local fillchars_as_text_width=1
+ local nb_fillchars=137
+ local i
+ (( i=0 ))
+ (( i < nb_fillchars ))
+ filled+='\[\]─'
+ (( i++  ))
+ (( i < nb_fillchars ))
+ filled+='\[\]─'
+ (( i++  ))
... # repeating forever
```

Reply:
I managed to reproduce, and it's not actually looping forever, but the call to `__lp_strip_escapes` is *extremely* slow.

This probably have to do with the fact that the passed `fillchars` is 137 characters, each of them being colored separately (albeit with the same color).

If you remove the `${lp_terminal_format}` line 691, it works fast and well.

As a fix, I am thinking of adding (yet another) argument to `_lp_fill` to allow for coloring the filled section with just one escaped tags.

Reply:
Fixable via [PR 803 on LP](https://github.com/liquidprompt/liquidprompt/pull/803)

Reply:
Now that liquidprompt/liquidprompt#803 is merged, this will need to be changed to not have the formatting in the fill string.